src/EventStore.ClientAPI/Projections/ProjectionsClient.cs

```diff
@@ -50,23 +52,41 @@ public Task CreateTransient(IPEndPoint endPoint, string name, string query, User
 
         public Task CreateContinuous(IPEndPoint endPoint, string name, string query, UserCredentials userCredentials = null)
         {
-            return SendPost(endPoint.ToHttpUrl("/projections/continuous?name={0}&type=JS&emit=1", name), 
+            return SendPost(endPoint.ToHttpUrl("/projections/continuous?name={0}&type=JS&emit=1", name),
                             query, userCredentials, HttpStatusCode.Created);
         }
 
-        public Task<string> ListAll(IPEndPoint endPoint, UserCredentials userCredentials = null)
+        public Task<List<ProjectionDetails>> ListAll(IPEndPoint endPoint, UserCredentials userCredentials = null)
         {
-            return SendGet(endPoint.ToHttpUrl("/projections/any"), userCredentials, HttpStatusCode.OK);
+            return SendGet(endPoint.ToHttpUrl("/projections/any"), userCredentials, HttpStatusCode.OK)
+                    .ContinueWith(x =>
+                    {
+                        if (x.IsFaulted) throw x.Exception;
+                        var r = JObject.Parse(x.Result);
+                        return r["projections"] != null ? r["projections"].ToObject<List<ProjectionDetails>>() : null;
+                    });
         }
 
-        public Task<string> ListOneTime(IPEndPoint endPoint, UserCredentials userCredentials = null)
+        public Task<List<ProjectionDetails>> ListOneTime(IPEndPoint endPoint, UserCredentials userCredentials = null)
         {
-            return SendGet(endPoint.ToHttpUrl("/projections/onetime"), userCredentials, HttpStatusCode.OK);
+            return SendGet(endPoint.ToHttpUrl("/projections/onetime"), userCredentials, HttpStatusCode.OK)
+                    .ContinueWith(x =>
+                    {
+                        if (x.IsFaulted) throw x.Exception;
+                        var r = JObject.Parse(x.Result);
+                        return r["projections"] != null ? r["projections"].ToObject<List<ProjectionDetails>>() : null;
+                    });
         }
 
-        public Task<string> ListContinuous(IPEndPoint endPoint, UserCredentials userCredentials = null)
+        public Task<List<ProjectionDetails>> ListContinuous(IPEndPoint endPoint, UserCredentials userCredentials = null)
```

Comment:
I think even changing it straight is actually ok projections are known to
be in flight. Also anyone using those methods would likely be quite happy
and have little exposure to the string based ones.

On Tue, Sep 8, 2015 at 7:57 PM, James Nugent <notifications@github.com>
wrote:

> In src/EventStore.ClientAPI/Projections/ProjectionsClient.cs
> <https://github.com/EventStore/EventStore/pull/665#discussion_r38958155>:
>
> >          }
> >
> > -        public Task<string> ListContinuous(IPEndPoint endPoint, UserCredentials userCredentials = null)
> > +        public Task<List<ProjectionDetails>> ListContinuous(IPEndPoint endPoint, UserCredentials userCredentials = null)
>
> Do we think we need to continue providing a Task<string> implementation
> (deprecated) for a while? @gregoryyoung <https://github.com/gregoryyoung>,
> @pgermishuys <https://github.com/pgermishuys>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EventStore/EventStore/pull/665/files#r38958155>.
>



-- 
Studying for the Turing test